### PR TITLE
Fix openid certificates on Azure App Services

### LINKED ISSUE #3222: OpenId token validation does not work when hosted in Azure App Service
All authenticated requests using a JWT bearer token fail when running in an Azure App Service.

Steps to reproduce:

Clone last OC, build and deploy to an Azure App service, then enable the OpenId Authorization Server module
image

Setup Open Id Authorization server to use JWT tokens
image

Setup an openId public client to connect using the implicit flow.
image

Use postman to a valid access token, then make a request to the /connect/userinfo endpoint.

Expected result:

Http 200 with JSON payload containing the Sub Id number

Actual result:

Http 401 Unauthorized
image

Doing these same setup steps and running locally, the request to /connect/userinfo works successfully, so it is something to do with running in Azure App Service that is causing the issue. This was working until recently.

### PR DESCRIPTION
**Pull request by sebastienros**
Fixes #3222

### CODE REVIEW

**Comment by sebastienros** [comment]
I tried with IIS express, and also checked /connect/user_info

File: src/OrchardCore.Modules/OrchardCore.OpenId/Services/OpenIdServerService.cs [hunk 1]
```diff
@@ -497,11 +497,14 @@ private async Task<ImmutableArray<(string path, X509Certificate2 certificate)>>
                                 X509KeyStorageFlags.EphemeralKeySet :
                                 X509KeyStorageFlags.MachineKeySet;
 #else
-                                X509KeyStorageFlags.MachineKeySet;
+                                X509KeyStorageFlags.MachineKeySet |
```
Comment (PinpointTownes) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Services/OpenIdServerService.cs]: Assuming you're going to target .NET Core, you'll likely hit the other branch (in the #ifdef). You may want to adopt the same code path for both .NET Standard and .NET Core, assuming we can make it work on every possible environment.

**Comment by PinpointTownes** [comment]
> I tried with IIS express, and also checked /connect/user_info

Sadly this is not enough: you'll need to test with full IIS with both user profile loading enabled and disabled (IIS Express always loads the user profile, AFAIK). You may want to test with both a virtual account (aka the application pool accounts) and a normal account, just to be sure.

**Comment by petedavis** [comment]
I'll try and test this on Azure. I have a feeling that it is the other #if branch that is compiled in.

If so we will have to remove the EphemeralKey code path and test elsewhere. That is the temp fix I did on our private branch.

**Comment by petedavis** [comment]
Hey @sebastienros this doesnt work on Azure. I will test my changes on full IIS, and if that seems OK then I'll update this pull request if I cna.

**Comment by PinpointTownes** [comment]
@petedavis I updated the PR to use a 2-pass approach that keeps the "don't persist the certificate private key anywhere on this machine" approach, but falls back to "persist it and mark it as exportable" if needed. Please give it a try when you have a minute.

**Comment by petedavis** [comment]
👍  Tested and working now on Azure App Services.

**Pull request merged**